telethon/client/chats.py

```diff
@@ -873,6 +873,12 @@ def action(
         )
 
         if isinstance(entity, types.InputPeerChannel):
+            if post_messages is None or edit_messages is None:
+                full_entity = await self.get_entity(entity)
+                if full_entity.megagroup:
+                    post_messages = None
```

Comment:
Could you add a comment as to why we do this? "If we try to set these permissions in a megagroup, we would get a `RIGHT_FORBIDDEN`. However, it makes sense that an admin can post messages, so we want to avoid the error".